extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -249,7 +249,9 @@ export class SubscriptionService {
       user().assert(this.pageConfig_, 'Page config is null');
 
       if (this.doesViewerProvideAuth_) {
-        return this.delegateAuthToViewer_();
+        this.delegateAuthToViewer_();
+        this.startAuthorizationFlow_();
```

Comment:
Overall this makes sense. But we can't actually activate the local platform. We are expecting all paywall features to be provided by the viewer in this scenario. The grant state stuff is good though.

Reply:
Ah ok so the dialog, action everything will be rendered by the viewer right?
In that case if it is verified we can just go ahead and call `this.processGrantState_(true)` manually.
WDYT?

Reply:
Either that, or you call just call this part:
```
    this.platformStore_.getGrantStatus()
        .then(grantState => {this.processGrantState_(grantState);});
```

Or you can add a boolean arg to `startAuthorizationFlow_` to disable the `selectAndActivatePlatform_` part.

Consider and see what looks like a better choice for now.